tools/install/windows/merge/BrokerMerge.wxs

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi"
+     xmlns:fire="http://schemas.microsoft.com/wix/FirewallExtension">
+
+  <?include ..\vars.wxi ?>
+
+  <Module Id="EtcRdmnetBroker" Language="0" Version="$(var.VersionNumber)" Codepage="1252">
+
+    <Package Id="DEE166EC-5441-43DC-9532-FFAFB81FFA93" Description="$(var.MergeDescription)"
+        Manufacturer="ETC Inc." InstallerVersion="200" Languages="1033" SummaryCodepage="1252"
+        InstallScope="perMachine" InstallPrivileges="elevated" />
+
+    <!-- ############### Broker Service Directory Structure ############### -->
+
+    <!-- The executable directory in Program Files (x86) -->
+    <Directory Id="TARGETDIR" Name="SourceDir">
+      <Directory Id="$(var.ProgramFilesFolder)">
+        <Directory Id="PROGRAM_FILES_ETC_DIR" Name="ETC">
+          <Directory Id="INSTALLDIR" Name="RDMnetBroker" FileSource="$(var.InstallSourceDir)">
+            <Component Id="Broker" Guid="2B4014E4-5278-469A-ABA7-A94F0AB850DA" SharedDllRefCount='yes'>
+              <File Id="BrokerExe" Name="RDMnetBrokerService.exe" DiskId="1" KeyPath="yes">
+                <!-- Create Windows Firewall exceptions for the Broker executable. -->
+                <fire:FirewallException Id="RDMnetBrokerTCP" Name="ETC RDMnet Broker (TCP-in)"
+                    Profile="all" Protocol="tcp" Scope="any" IgnoreFailure="no" />
+                <fire:FirewallException Id="RDMnetBrokerUDP" Name="ETC RDMnet Broker (UDP-in)"
+                    Profile="all" Protocol="udp" Scope="any" IgnoreFailure="no" />
+              </File>
+              <File Id="DnsSdDll" Name="dnssd.dll" DiskId="1" />
+              <ServiceControl Id="BrokerServiceCtl" Name="ETC RDMnet Broker" Stop="both"
+                  Start="install" />
+            </Component>
+          </Directory>
+        </Directory>
+      </Directory>
+
+      <!-- The ProgramData directory for the conf and log files -->
+      <Directory Id="CommonAppDataFolder">
+        <Directory Id="PROGRAM_DATA_ETC_DIR" Name="ETC">
+          <Directory Id="PROGRAM_DATA_RDMNET_BROKER_DIR" Name="RDMnetBroker" FileSource="src">
+            <Component Id="BrokerConf" Guid="C24F6AF3-1DCD-4160-A604-200E78401721">
+              <File Id="BrokerConfFile" Name="broker.conf" DiskId="1" KeyPath="yes" />
+            </Component>
+          </Directory>
+        </Directory>
+      </Directory>
+    </Directory>
+
+    <!-- ######################### Custom Actions ######################### -->
+
+    <!--
+    The RDMnetBrokerService executable is invoked with the argument "-install"
+    on installation and "-remove" on uninstallation - this does the actual
+    registration and de-registration of the service with the Service Control
+    Manager.
+    -->
```

Comment:
Is this service registration unique to your service or did you find that this was the generally accepted way to go about this? I've not had to do services in an MSI yet so I'm curious.

Reply:
This method (complete with the `-install` and -remove` flags) seems to be the generally accepted practice in almost every service I've come across:
- The example service code available on MSDN
- Apple's Bonjour for Windows service
- Our SLP and SNTP service